src/actions/public/L2/feeActions.ts

```diff
@@ -0,0 +1,204 @@
+import { gasPriceOracleABI, gasPriceOracleAddress } from '@eth-optimism/contracts-ts'
+import { type Abi } from 'abitype'
+import {
+  Address,
+  BlockTag,
+  encodeFunctionData,
+  EncodeFunctionDataParameters,
+  EstimateGasParameters,
+  getContract,
+  PublicClient,
+  serializeTransaction,
+  TransactionSerializableEIP1559,
+  TransactionSerializedEIP1559,
+  Transport,
+} from 'viem'
+import { type Chain } from 'viem/chains'
+
+/**
+ * Bytes type representing a hex string with a 0x prefix
+ */
+type Bytes = `0x${string}`
+
+/**
+ * Options to query a specific block
+ */
+type BlockOptions = {
+  /**
+   * Block number to query from
+   */
+  blockNumber?: bigint
+  /**
+   * Block tag to query from
+   */
+  blockTag?: BlockTag
+}
+
+/**
+ * Options for all GasPriceOracle methods
+ */
+export type GasPriceOracleParameters = BlockOptions
+
+/**
+ * Options for specifying the transaction being estimated
+ */
+export type OracleTransactionParameters<
+  TAbi extends Abi | readonly unknown[],
+  TFunctionName extends string | undefined = undefined,
+> =
+  & EncodeFunctionDataParameters<TAbi, TFunctionName>
+  & Omit<TransactionSerializableEIP1559, 'data' | 'type'>
+  & GasPriceOracleParameters
+/**
+ * Options for specifying the transaction being estimated
+ */
+export type GasPriceOracleEstimator = <
+  TChain extends Chain | undefined,
+  TAbi extends Abi | readonly unknown[],
+  TFunctionName extends string | undefined = undefined,
+>(
+  client: PublicClient<Transport, TChain>,
+  options: OracleTransactionParameters<TAbi, TFunctionName>,
+) => Promise<bigint>
+
+/**
+ * Internal helper to serialize a transaction
+ */
+const transactionSerializer = <
+  TAbi extends Abi | readonly unknown[],
+  TFunctionName extends string | undefined = undefined,
+>(
+  options:
+    & EncodeFunctionDataParameters<TAbi, TFunctionName>
+    & Omit<TransactionSerializableEIP1559, 'data'>,
+): TransactionSerializedEIP1559 => {
+  const encodedFunctionData = encodeFunctionData(options)
+  const serializedTransaction = serializeTransaction({
+    ...options,
+    data: encodedFunctionData,
+    type: 'eip1559',
+  })
+  return serializedTransaction as TransactionSerializedEIP1559
+}
+
+/**
+ * Get gas price Oracle contract
+ * @example
+ * const contract = getGasPriceOracleContract(publicClient)
+ * const gasPrice = await contract.read.gasPrice({ blockNumber })
+ */
+const getGasPriceOracleContract = <TChain extends Chain | undefined>(
```

Comment:
again prefer 1 function per file. 

Thinking if we should actually export these viem versions of all the main contracts 🤔

Reply:
@evmts/contracts-ts is the package that exports versions of all the main contracts. It does it both as react hooks and as vanilla hooks. It might be a good idea to reexport them from this package though